### PR TITLE
Add rlang style argument checking to all functions

### PR DESCRIPTION
**Pull request by hadley**

### CODE REVIEW

File: R/collapse.R [hunk 1]
```diff
@@ -19,6 +19,7 @@
 #' fct_count(partyid2)
 fct_collapse <- function(.f, ..., other_level = NULL, group_other = "DEPRECATED") {
   f <- check_factor(.f)
+  check_string(other_level, allow_null = TRUE, allow_na = TRUE)
```
Comment (DavisVaughan) [inline R/collapse.R:22]: Do we allow empty strings for levels?

File: R/expand.R [hunk 2]
```diff
@@ -15,6 +15,7 @@
 fct_expand <- function(f, ..., after = Inf) {
   f <- check_factor(f)
   check_dots_unnamed()
+  check_number_decimal(after)
```
Comment (DavisVaughan) [inline R/expand.R:18]: Minimum value of `0` i believe?

Also is it `check_number_whole()`?

File: R/lump.R [hunk 3]
```diff
@@ -87,25 +87,21 @@ fct_lump <- function(f, n, prop, w = NULL, other_level = "Other",
 #' @rdname fct_lump
 fct_lump_min <- function(f, min, w = NULL, other_level = "Other") {
   f <- check_factor(f)
-  level_w <- compute_weights(f, w)
-
-  if (!is.numeric(min) || length(min) != 1 || min < 0) {
-    cli::cli_abort("{.arg min} must be a positive number.")
-  }
+  check_number_decimal(min, min = 0)
```
Comment (DavisVaughan) [inline R/lump.R:90]: whole vs decimal?

File: R/shift.R [hunk 4]
```diff
@@ -18,6 +18,7 @@
 #' fct_shift(x, 2)
 #' fct_shift(x, -1)
 fct_shift <- function(f, n = 1L) {
```
Comment (DavisVaughan) [inline R/shift.R:20]: `check_factor()` here?

**Review by DavisVaughan — APPROVED** [review]

File: R/collapse.R (hunk 1 above)
Comment (hadley) [reply R/collapse.R:22]: Hmmm, technically I guess we should.

File: R/expand.R (hunk 2 above)
Comment (hadley) [reply R/expand.R:18]: `check_number_whole(Inf)` throws an error 😞

**Pull request merged**